src/Sarif/Writers/PrereleaseCompatibilityTransformer.cs

```diff
@@ -0,0 +1,553 @@
+            modifiedVersionControlDetails |= RenameProperty(versionControlDetails, previousName: "uri", newName: "repositoryUri");
+
+            // https://github.com/oasis-tcs/sarif-spec/issues/249
+            modifiedVersionControlDetails |= RenameProperty(versionControlDetails, previousName: "tag", newName: "revisionTag");
+
+            return modifiedVersionControlDetails;
+        }
+
+        internal static bool UpdateRunResults(JArray runs)
+        {
+            bool modifiedRun = false;
+
+            foreach (JObject run in runs)
+            {
+                JArray results = (JArray)run["results"];
+
+                if (results == null) { continue; }
+
+                foreach (JObject result in results)
+                {
+                    // https://github.com/oasis-tcs/sarif-spec/issues/216
+                    //
+                    // result.ruleMessageId is deprecated. This property value
+                    // should be expressed as result.message.messageId instead.
+                    // e.g.:
+                    // "message": { "messageId" : "default"}
+
+                    string ruleMessageId = (string)result["ruleMessageId"];
+
+                    if (ruleMessageId != null)
+                    {
+                        result.Property("ruleMessageId").Remove();
+                        result["message"]["messageId"] = ruleMessageId;
+                        modifiedRun = true;
+                    }
+
+                    var codeFlows = (JArray)result["codeFlows"];
+                    if (codeFlows != null)
+                    {
+                        modifiedRun |= UpdateCodeFlows(codeFlows);
+                    }
+                }
+            }
+            return modifiedRun;
+        }
+
+        private static bool UpdateCodeFlows(JArray codeFlows)
+        {
+            bool modifiedThreadFlowLocation = false;
+
+            foreach (JObject codeFlow in codeFlows)
+            {
+                JArray threadFlows = (JArray)codeFlow["threadFlows"];
+                if (threadFlows == null) { continue; }
+
+                foreach (JObject threadFlow in threadFlows)
+                {
+                    JArray threadFlowLocations = (JArray)threadFlow["locations"];
+                    if (threadFlowLocations == null) { continue; }
+
+                    foreach (JObject threadFlowLocation in threadFlowLocations)
+                    {
+                        // Update name of threadFlowLocation.executionTimeUtc: https://github.com/oasis-tcs/sarif-spec/issues/242
+                        modifiedThreadFlowLocation |= RenameProperty(threadFlowLocation, previousName: "timestamp", newName: "executionTimeUtc");
+
+                        // Delete threadFlowLocation.step property: https://github.com/oasis-tcs/sarif-spec/issues/203
+                        JToken step = threadFlowLocation["step"];
+                        if (step?.Type == JTokenType.Integer)
```

Comment:
>if (step?.Type == JTokenType.Integer) [](start = 24, length = 37)

Why the test? You wouldn't keep it around if it were (for example) a string, right?